var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -61,9 +61,19 @@ class Esmf(MakefilePackage):
     # Testing dependencies
     # depends_on('perl', type='test')  # TODO: Add a test deptype
 
-    # NOTE: ESMF cannot be installed with GCC 6. It uses constructs that
-    # are no longer valid in GCC 6. GCC 4 is recommended for installation.
-    conflicts('%gcc@6:')
+    # Make esmf build with newer gcc versions
+    # https://sourceforge.net/p/esmf/esmf/ci/3706bf758012daebadef83d6575c477aeff9c89b/
+    patch('gcc.patch', when='%gcc@6:')
+
+    # Fix undefined reference errors with mvapich2
+    # https://sourceforge.net/p/esmf/esmf/ci/34de0ccf556ba75d35c9687dae5d9f666a1b2a18/
+    patch('mvapich2.patch')
```

Comment:
If both of the got merged, you might want to add a `when='@7.0.1'`!

Reply:
And this patch is pretty big, so is there are a way to turn it into and url patch?

Reply:
I added the `when` clauses but could not find a way to get the raw patch from SourceForge's web interface.